msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/AccountState.kt

```diff
@@ -220,21 +236,71 @@ class AccountState private constructor(
     }
 
     /**
-     * Retrieves the access token for the currently signed in account from the cache.
+     * Retrieves the access token for the currently signed in account from the cache. If multiple
+     * access tokens are present in the cache then the one that matches the sign in scopes is returned.
      * If the access token is expired, it will be attempted to be refreshed using the refresh token that's stored in the cache;
      * Kotlin coroutines variant.
      *
      * @return [com.microsoft.identity.nativeauth.statemachine.results.GetAccessTokenResult] The result of the getAccessToken action
      */
     suspend fun getAccessToken(forceRefresh: Boolean = false): GetAccessTokenResult {
+        return getAccessTokenInternal(forceRefresh, null);
+    }
+
+    /**
+     * Retrieves the access token for the currently signed in account from the cache.
+     * If the access token is expired, it will be attempted to be refreshed using the refresh token that's stored in the cache;
+     * If any of the cached access token matches the scope specified in scopes parameter then that
+     * is returned otherwise a new access token with requested scopes is fetched.
+     * Kotlin coroutines variant.
+     *
+     * @return [com.microsoft.identity.nativeauth.statemachine.results.GetAccessTokenResult] The result of the getAccessToken action
+     */
+    suspend fun getAccessToken(forceRefresh: Boolean = false, scopes: List<String>?): GetAccessTokenResult {
```

Comment:
I think the scopes list should not be optional here

Reply:
Making it null allows the developers to move to this function from the old deprecated method.

Reply:
@nilo-ms can you confirm what should happen here? Is scopes optional, or not?
- in current MSAL, it's not optional (i.e. if an empty scopes list is passed, the method returns an error)
- if we allow empty/null scopes param, how do we communicate to developers how this method works in those scenarios?
- how is iOS implementing this for native auth? 
- I agree with Saurabh that developers need a way to migrate from the old, now-deprecated method to the new one. If the old one doesn't take a scopes parameter, then the new one needs to have the possibility of also not passing any.

Reply:
I think native auth should work in the same way as the platform specific MSAL works. For example, for MSAL Objective-C, the scope list is not optional, but the external developer can specify an empty list of scopes, no error is thrown. This is how we plan to implement it.
Given that you want to provide a way to migrate to the new method, I suggest to slightly deviate from how MSAL Android work. I propose:
- adding the scope list as mandatory
- If an empty scope list is passed by the external developer, add the default OIDC scopes internally (as we do for the signIn) 
- clarify this behaviour in the method comment
Be mindful that if you want to align with the MSAL Android behaviour, it will be considered a breaking change.

Reply:
I have made the parameter as NonNull

Reply:
@SaurabhMSFT can you clarify which approach you're taking:
1. conceptually, scopes are optional. Similar to how getAccessToken() behaves right now.
2. scopes are not optional. Similar to how MSAL browser-delegated behaves right now.

If we're going with 1, then the following should be implemented IMO:
1. scopes param may be `null`.
2. scopes list may be empty.

As these two are effectively the same, I don't see a reason to limit one but not the other.

Reply:
Scopes are optional. An empty list of scopes or a null value could show the same intent. Supporting one or the other is just a matter of taste and consistency.

Reply:
I disagree @SaurabhMSFT. If scopes are optional, the whole parameter should be nullable. It's a pain for developers to have to create an empty list, rather than be allowed to pass `null`. both `null` and empty list serve the same functional purpose, so both should be allowed.